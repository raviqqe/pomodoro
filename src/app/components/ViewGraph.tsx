import { IoMdStats } from "react-icons/io";
import { CircleButton } from "./CircleButton.js";

interface Props {
  onClick: () => void;
}

export const ViewGraph = ({ onClick: viewGraph }: Props): JSX.Element => (
  <CircleButton aria-label="View Graph" onClick={viewGraph} secondary>
    <IoMdStats />
  </CircleButton>
);
