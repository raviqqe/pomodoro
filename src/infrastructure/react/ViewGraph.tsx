import { IoMdStats } from "react-icons/io";
import { CircleButton } from "./CircleButton";

interface IProps {
  viewGraph: () => void;
}

export const ViewGraph = ({ viewGraph }: IProps): JSX.Element => (
  <CircleButton aria-label="View Graph" onClick={viewGraph} secondary={true}>
    <IoMdStats />
  </CircleButton>
);
