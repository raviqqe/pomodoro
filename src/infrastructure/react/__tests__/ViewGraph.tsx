import { create } from "react-test-renderer";
import { ViewGraph } from "../ViewGraph";

it("renders", () => {
  expect(create(<ViewGraph viewGraph={() => {}} />).toJSON()).toMatchSnapshot();
});
