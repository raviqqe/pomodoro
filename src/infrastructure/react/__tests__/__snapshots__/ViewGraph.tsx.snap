// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<button
  aria-label="View Graph"
  className="sc-bdfBQB sc-gsTEea gSdseo jGCERI"
  onClick={[Function]}
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M176 64h64v384h-64zM80 336h64v112H80zM272 272h64v176h-64zM368 176h64v272h-64z"
    />
  </svg>
</button>
`;
